File: backend/app/schemas/master_setlist.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional, List


class MasterSetlistCreate(BaseModel):
    band_id: int
    name: str
    description: Optional[str] = None


class MasterSetlistUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None


class MasterSetlistResponse(BaseModel):
    id: int
    band_id: int
    name: str
    description: Optional[str] = None
    is_active: bool
    created_at: datetime
    updated_at: datetime
    song_count: Optional[int] = 0
    
    class Config:
        from_attributes = True


class MasterSetlistWithSongsResponse(MasterSetlistResponse):
    songs: List["SongBriefResponse"] = []


# Brief song response for nested use (avoids circular import)
class SongBriefResponse(BaseModel):
    id: int
    title: str
    scale: Optional[str] = None
    genre: Optional[str] = None
    position: Optional[int] = 0
    
    class Config:
        from_attributes = True


# Update forward reference
MasterSetlistWithSongsResponse.model_rebuild()

File: frontend/src/types/masterSetlist.ts
export interface MasterSetlist {
  id: number;
  band_id: number;
  name: string;
  description?: string | null;
  is_active: boolean;
  created_at: string;
  updated_at: string;
  song_count?: number;
}

export interface MasterSetlistCreate {
  band_id: number;
  name: string;
  description?: string | null;
}

export interface MasterSetlistUpdate {
  name?: string;
  description?: string | null;
}

export interface SongBrief {
  id: number;
  title: string;
  scale?: string | null;
  genre?: string | null;
  position?: number;
}

export interface MasterSetlistWithSongs extends MasterSetlist {
  songs: SongBrief[];
}